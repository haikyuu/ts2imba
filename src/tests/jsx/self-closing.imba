import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
function App(){
	return (
		<div className="bg-gray-200">
			<div>
				<img src="" />
			</div>
			<span>Test</span>
		</div>
	)
}
'''

test 'self closing 11' do
	const result = await build tsx-code
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')
