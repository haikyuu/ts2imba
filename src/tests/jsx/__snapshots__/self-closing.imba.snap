// Vitest Snapshot v1

exports[`self closing 11 1`] = `
"tag App
	<self>
		<div [bgc:gray2]>
			<div>
				<img src=\\"\\">
			<span>
				\\"Test\\"
"
`;
